src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -720,208 +636,123 @@ void OpticalTactilePluginPrivate::DepthCameraCallback(
-        // end of the cylinder in the surface, not its middle point
-        ignition::math::Pose3f forceMarkerSensorPose(
-            forceMarkerSensorPosition +
-            normalForces[i][j] * (this->forceLength/2),
-            forceMarkerSensorQuaternion);
-
-        ignition::math::Pose3f forceMarkerWorldPose =
-            (forceMarkerSensorPose + this->depthCameraOffset) +
-            this->sensorWorldPose;
-        forceMarkerWorldPose.Correct();
-
-        ignition::math::Pose3f positionMarkerSensorPose(
-            forceMarkerSensorPosition, forceMarkerSensorQuaternion);
-        ignition::math::Pose3f positionMarkerWorldPose =
-            (positionMarkerSensorPose + this->depthCameraOffset) +
-            this->sensorWorldPose;
-        positionMarkerWorldPose.Correct();
-
-        // Set markers pose messages from previously computed poses
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_position()->set_x(forceMarkerWorldPose.Pos().X());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_position()->set_y(forceMarkerWorldPose.Pos().Y());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_position()->set_z(forceMarkerWorldPose.Pos().Z());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_x(forceMarkerWorldPose.Rot().X());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_y(forceMarkerWorldPose.Rot().Y());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_z(forceMarkerWorldPose.Rot().Z());
-        this->forceMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_w(forceMarkerWorldPose.Rot().W());
-
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_position()->set_x(positionMarkerWorldPose.Pos().X());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_position()->set_y(positionMarkerWorldPose.Pos().Y());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_position()->set_z(positionMarkerWorldPose.Pos().Z());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_x(positionMarkerWorldPose.Rot().X());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_y(positionMarkerWorldPose.Rot().Y());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_z(positionMarkerWorldPose.Rot().Z());
-        this->positionMarkerMsg.mutable_pose()
-          ->mutable_orientation()->set_w(positionMarkerWorldPose.Rot().W());
-
-        this->node.Request("/marker", this->forceMarkerMsg);
-        this->node.Request("/marker", this->positionMarkerMsg);
-      }
-      visualizeColumn = (this->resolution == ++columnCounter);
-      if (visualizeColumn)
-          columnCounter = 0;
+      markerPosition = this->MapPointCloudData(i, j);
+      this->visualizePtr->VisualizeNormalForce(i, j, markerPosition,
```

Comment:
About `/list`, check this comment: https://github.com/ignitionrobotics/ign-gazebo/pull/250#discussion_r467732754

Looking at what the `/marker` service [does](https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/rendering/MarkerManager.cc#L592-L597), it shouldn't be very difficult to implement that.

Anyway, should I make an issue + TODO comment?

Reply:
Ah, sorry I did not look at the method body closely for `/list`. Looks like Ignition doesn't have the capability to publish an array of markers.

However, @chapulina has suggested using lines and points, instead of cylinders and spheres. https://github.com/ignitionrobotics/ign-msgs/blob/6b4430a91e39e66ad6e9c2844c824e36aebc8b48/proto/ignition/msgs/marker.proto#L147-L154

LINE_LIST and POINTS probably do the same as their counterparts in RViz and should give you similar results. They are essentially arrays, but they only need to be published once. You would `push_back` each point into the vector, and only publish once. http://wiki.ros.org/rviz/DisplayTypes/Marker#Line_List_.28LINE_LIST.3D5.29

Reply:
> LINE_LIST and POINTS probably do the same as their counterparts in RViz and should give you similar results. They are essentially arrays, but they only need to be published once

I've tried that, and it makes a huge difference. I don't know if that difference comes from transporting only one message or rendering a simpler figure though.

> Looking at what the /marker service does, it shouldn't be very difficult to implement the service we're looking for.
Should I make an issue + TODO comment?

I think that I should do this as well, it could be a nice `good first issue`. Besides, if the performance using cylinders and spheres is the same, we should probably use those because it looks way cooler.

Here's a screenshot with LINE_LIST and POINTS. Btw, distinguishing the points is really difficult, so I'll probably remove those:

![line_list](https://user-images.githubusercontent.com/14854080/89909840-cc54d600-dbef-11ea-92f8-582a2ccca9f5.png)

Reply:
Both. The simpler geometry certainly makes a difference, but the reduction in messages as well. That is my experience with RViz. The performance would not be the same with cylinders and spheres, because they are each rendering many faces. Round geometry especially take longer. I think it is actually good to keep it lines and points for this plugin, because that will enable adding many more plugins. I didn't think of that at the beginning but that is where we should have started.

Try making the point size larger. In RViz, the points are rendered as a single square, and as you rotate the view, you can see them better. Don't remove them, because there are cases when normals may be invalid and the points are the only information the user can go by.

Feel free to create an issue, but we don't need it for this plugin.

Reply:
> Try making the point size larger

Hmm I don't know if this is currently possible. I've tried the `scale` field, but all it does is changing the pose of the marker, but the size is always the same. The `pose` field also changes the pose of the marker.

Reply:
Sounds like a bug, either in Ignition or in the plugin. I looked for where the marker rendering code sets the scale, and it looks very simple, one block handling all types of markers https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/rendering/MarkerManager.cc#L255

Maybe add a printout from the `has_scale` block and `has_pose` block there and see if it's getting the values you input? Then you can go from there to see where the bug is. If you find the bug is in Ignition, we should open an issue.

Reply:
> Maybe add a printout from the has_scale block and has_pose block there and see if it's getting the values you input? 

The data passed is the same that I set. I couldn't find the issue though, but it must be in `ign-rendering`.
The rest of these comment is addressed in e216bb2